### PR TITLE
Photoshop: fix layer publish thumbnail missing in loader

### PR DESCRIPTION
**Pull request by kalisp**
## Changelog Description
Thumbnails from any products (either `review` nor separate layer instances) weren't stored in Ayon.
This resulted in not showing them in Loader and Server UI. After this PR thumbnails should be shown in the Loader and on the Server (`http://YOUR_AYON_HOSTNAME:5000/projects/YOUR_PROJECT/browser`).

## Additional info
Tis PR explicitly sets path to thumbnail on the instance (as there might be only one thumbnail per instance, imho) bypassing issue that representation with 'name' == 'thumbnail' is by default set to not be integrated.

## Testing notes:
1. publish workfile in PS with default `review` instance
2. publish workfile in PS with separate `image` instances which have `Review` toggle set to True 
image
3. Check right column in the Loader and Server UI

### CODE REVIEW

**Review by 64qam — APPROVED** [review]

File: openpype/plugins/publish/integrate_thumbnail_ayon.py [hunk 1]
```diff
@@ -5,7 +5,19 @@
     pull into a scene.
 
     This one is used only as image describing content of published item and
-    shows up only in Loader in right column section.
+    shows up only in Loader or Server UI in right column section.
+
+    Possible sources of thumbnail paths (OR):
+    - context.data.get("thumbnailPath")
+    (instance must have "published_representations")
+    - instance.data.get("thumbnailSource")
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_thumbnail_ayon.py:13]: This should not be used https://github.com/ynput/OpenPype/pull/6063
```suggestion
```

File: openpype/hosts/photoshop/plugins/publish/extract_review.py [hunk 2]
```diff
@@ -224,6 +224,7 @@ def _generate_thumbnail(
             "stagingDir": staging_dir,
             "tags": ["thumbnail", "delete"]
         })
+        instance.data["thumbnailPath"] = thumbnail_path  # for Ayon integration
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/plugins/publish/extract_review.py:227]: It is used for OpenPype too
```suggestion
        instance.data["thumbnailPath"] = thumbnail_path
```

File: openpype/plugins/publish/integrate_thumbnail_ayon.py [hunk 3]
```diff
@@ -5,7 +5,19 @@
     pull into a scene.
 
     This one is used only as image describing content of published item and
-    shows up only in Loader in right column section.
+    shows up only in Loader or Server UI in right column section.
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_thumbnail_ayon.py:8]:
```suggestion
    shows up only in Loader or WebUI in right column section.
```

File: openpype/plugins/publish/integrate_thumbnail_ayon.py [hunk 4]
```diff
@@ -5,7 +5,19 @@
...
+
+    Possible sources of thumbnail paths (OR):
+    - context.data.get("thumbnailPath")
+    (instance must have "published_representations")
+    - instance.data.get("thumbnailSource")
+    - instance.data.get("thumbnailPath")
+    - representation with 'thumbnail' name from "published_representations"
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_thumbnail_ayon.py:15]: thumbnailSource should not be used https://github.com/ynput/OpenPype/pull/6063
```suggestion
    Instance must have "published_representations" to
        be able to integrate thumbnail.
    Possible sources of thumbnail paths:
    - instance.data["thumbnailPath"]
    - representation with 'thumbnail' name from "published_representations"
    - context.data["thumbnailPath"]
    
```

File: openpype/plugins/publish/integrate_thumbnail_ayon.py [hunk 5]
```diff
@@ -5,7 +5,19 @@
...
+
+    (issue with last option is that we most likely don't want to integrate
+    thumbnail, eg. store it DB. Integrated representation would be polluting
+    Loaader (and DB) and its use is likely minimal.)
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_thumbnail_ayon.py:19]:
```suggestion
    Note:
        Issue with thumbnail representation is that we most likely don't
            want to integrate it as representation. Integrated representation
            is polluting Loader and databse without real usage.
```

File: openpype/plugins/publish/integrate_thumbnail_ayon.py (hunk 1 above)
Comment (iLLiCiTiT) [reply openpype/plugins/publish/integrate_thumbnail_ayon.py:13]: Replaced by this comment https://github.com/ynput/OpenPype/pull/6061/files#r1429866462

**Pull request merged**